cmd/launcher/gui/gui_progress_control.go

```diff
@@ -138,7 +139,7 @@ func updateProgressLabel() {
 			delta := panelDownloadStatus.progressCurrent - panelDownloadStatus.progressPrevious
 			var message string
 			if panelDownloadStatus.stage.IsDownloadStage() {
-				message = fmt.Sprintf("Downloading at %s. ", rateString(delta))
+				message = fmt.Sprintf("Downloading at %s. ", rateString(uint64(float64(delta) / labelUpdateInterval.Seconds())))
```

Comment:
This is the only case that this method is called. As most of the method is using float64 anyways I would suggest instead of casting to change the method parameter type.

Reply:
Yup! Refactored the method and added a test!